console.lua

```diff
@@ -97,3 +101,107 @@ function BuffDuty:convertPlayerList(identifier, input)
     end
     return result
 end
+
+function Console.parseMessageCommand(cmd, ...)
+    local arg = {...} -- Arg list
+    local idx = 1 -- Current Arg index
+
+    local function arg_valid(i)
+        i = i or idx
+        return (arg[i] and not (i == #arg)) -- Not nil, and ignore the last Arg that AceConsole appends
+    end
+    
+    -- Print Usage Help
+    if arg[1] == "?" or arg[1] == "help" then
+        BuffDuty.printInfoMessage("Usage: /buffduty-msg [options]")
+        BuffDuty.printInfoMessage("reset | Resets all messages to default values")
+        BuffDuty.printInfoMessage("public-title \"value\" | Sets public title message format to \"value\"")
+        BuffDuty.printInfoMessage("duty-line \"value\" | Sets duty line message format to \"value\"")
+        BuffDuty.printInfoMessage("duty-whisper \"value\" | Sets duty whisper message format to \"value\"")
+        BuffDuty.printInfoMessage("single-title \"value\" | Sets single title message format to \"value\"")
+        BuffDuty.printInfoMessage("single-whisper \"value\" | Sets single whisper message format to \"value\"")
+        return false
+    end
+    
+    -- Options
+    while arg_valid() do
+        local tag = string.match(arg[idx], "^[%a%-]+") -- Match starting letters including '-' and '?'
+        -- Reset All
+        if tag == "reset" then
+            cmd.reset_all = true
+        -- Public Title
+        elseif tag == "public-title" or tag == "-pt" then
```

Comment:
@Byron-Miles let's move these parts to a separate functions. I bet we can make 1 generic function and use it without code duplication.

Reply:
I was thinking of doing something like that, even to the point of putting the functions in a  table where the tags are the keys... this would reduce the loop down to something like `fn = table[tag] if fn then idx = fn(tag, arg, idx) end` ... I'll have a play with it and see if I can work out a nice generic pattern.

Reply:
I've restructed this part of console.lua ... it now uses a generic executeOptionalArgs function which takes a table indexed by tag, For now the table is still built in parseMessageCommand, but that will probably change when the new console parsing is implimented for the base /buffduty command